### ISSUE TITLE
Download debs instead of compiling from source

### ISSUE THREAD

**Issue by hamiltont** [TechEmpower/FrameworkBenchmarks#1479]
This will be much, much faster than compiling big projects from source, doesn't require sudo, and still installs everything into a prefix folder. I expect we can also find a version of this that works with rpm-based systems. 

Turns out this is really easy to do. Here's a rough pattern: 

```
# Download and install without sudo
apt-get download somepackage
dpkg -x somepackage_*.deb $IROOT/somepackage

# Binaries and Libraries are now stores inside here
# For each somepackage you have to ID where they are precisely
# so you can use them
SOMEROOT=$IROOT/somepackage/usr/local/bin/somepackage
```

Note that `apt-get download` does not also download dependencies. If you need to download dependencies too, use this apt command instead to download all the necessary deb files. 

```
function apt-download { wget -c $(apt-get install --reinstall --print-uris -qq $1 | cut -d"'" -f2); }
```

Thoughts?

**Comment by hamiltont** [comment]
Function credit: http://askubuntu.com/a/30581/91562

**Comment by msmith-techempower** [comment]
I think that it's an interesting idea, but it ultimately doesn't solve the problem that is all `apt-get` are not created equal. For instance, our suite doesn't install properly on Debian because the package names for `apt-get` aren't 100% equal.

The only real solution I can offer is to scour for permalinks to individual `.deb` files as needed, then do the `dpkg` stuff. Yes, this will require each installation to know its own dependencies, but that is probably even better -- we'll know exactly what is required for a given stack.

**Comment by ssmith-techempower** [comment]
I've run into issues with `apt-get` while getting TFB running on our Raspberry Pi 2. Apt will potentially install things in platform-specific locations, which leads to issues caused by frameworks' setup scripts looking for things in platform-specific locations.

The Java frameworks specifically seem to assume `java` is located at:

    /usr/lib/jvm/java-1.7.0-openjdk-amd64/bin/java

On the RPi2 with their flavor of Ubuntu 14.04 `apt-get` installs it here:

    /usr/lib/jvm/java-1.7.0-openjdk-armhf/bin/java

**Comment by nbrady-techempower** [comment]
I think this can be closed for #1500 though I still think we're a ways away from that. I'm working on travis and toolset changes currently, and will see if I can start moving things toward distro agnostic installations.
